src/UbiHttp.cpp

```diff
@@ -0,0 +1,375 @@
+
+  _client_https_ubi.stop();
+  return result;
+}
+
+/**
+ * @brief Calculate the lenght of the request line to be send over HTTP to the
+ * server
+ *
+ * @param device_label label set by the user
+ * @param payload  request built by the Library
+ * @param pathLength Lengh of the request line
+ * @return uint16_t Total length of the request
+ */
+uint16_t UbiHTTP::_buildRequestLength(const char *device_label,
+                                      const char *payload,
+                                      uint16_t pathLength) {
+  uint16_t endpointLength =
+      strlen("POST  HTTP/1.1\r\n"
+             "Host: \r\n"
+             "User-Agent: \r\n"
+             "X-Auth-Token: \r\n"
+             "Connection: close\r\n"
+             "Content-Type: application/json\r\n"
+             "Content-Length: \r\n"
+             "\r\n"
+             "\r\n") +
+      pathLength + strlen(device_label) + strlen(_host) + strlen(_user_agent) +
+      strlen(_token) + UbiUtils::countDigit(strlen(payload)) + strlen(payload);
+
+  return endpointLength;
+}
+
+double UbiHTTP::get(const char *device_label, const char *variable_label) {
+  if (_debug) {
+    Serial.print(F("Connecting to "));
+    Serial.println(_host);
+  }
+
+  if (!_client_https_ubi.connectSSL(_host, _port)) {
+    if (_debug) {
+      Serial.println(F("Connection Failed to Ubidots - Try Again"));
+    }
+    if (!reconnect<WiFiSSLClient>(&_client_https_ubi)) {
+      return ERROR_VALUE;
+    }
+  }
+
+  uint16_t pathLength = _pathLength(device_label, variable_label);
+  char *path = (char *)malloc(sizeof(char) * pathLength + 1);
+  sprintf(path, "/api/v1.6/devices/%s/%s/lv", device_label, variable_label);
+
+  if (_debug) {
+    Serial.print(F("Requesting to URL: "));
+    Serial.println(path);
+  }
+
+  uint16_t requestLineLength = _requestLineLength(path);
+  char *message = (char *)malloc(sizeof(char) * requestLineLength + 1);
+  sprintf(message,
+          "GET %s HTTP/1.1\r\n"
+          "Host: %s\r\n"
+          "X-Auth-Token: %s\r\n"
+          "User-Agent: %s\r\n"
+          "Content-Type: application/json\r\n"
+          "Connection: close\r\n"
+          "\r\n",
+          path, _host, _token, _user_agent);
+
+  if (_debug) {
+    Serial.println(F("Request sent"));
+    Serial.println(message);
+  }
+
+  _client_https_ubi.print(message);
+
+  while (_client_https_ubi.connected()) {
+    String line = _client_https_ubi.readStringUntil('\n');
+    if (line == "\r") {
+      if (_debug) {
+        Serial.println(F("Headers received"));
+      }
+      break;
+    }
+  }
+
+  free(message);
+  free(path);
+
+  double value = _parseServerAnswer();
+  _client_https_ubi.flush();
+  _client_https_ubi.stop();
+  // return 0;
+  return value;
+}
+
+double UbiHTTP::_parseServerAnswer() {
+
+  /**
+   * @param _charResponseLength 3 is the maximun amount of digits for the
+   * length to have
+   */
+  char *_charLength = (char *)malloc(sizeof(char) * 3);
+
+  readServerAnswer(_charLength);
```

Comment:
what happens here if the variable or the device does not existe in the account?

Reply:
Did think about it....